src/actions/ProductsActions.ts

```diff
@@ -64,3 +69,26 @@ export const getProductsByProvider = (id: string, query = '') => {
       });
   });
 };
+
+export const getProductById = (id: string) => {
+  return new Promise<product>((resolve, reject) => {
+    useQuery<getProductByIdReturnTypes>(getProductByIdQuery, { id })
+      .then(({ products }) => {
+        if (!products || !products[0]) {
```

Comment:
please seak to use length here
```suggestion
        if (!products || !products.length) {
```

I am not sure if `{}` is truthy

Reply:
Will do, Thanks for the suggestion

Reply:
hmm I think it's truthy

```!!{} === true```

but your suggestion is better, thanks!